lib/Optimizer/Transforms/QuakeSynthesizer.cpp

```diff
@@ -543,17 +546,24 @@ class QuakeSynthesizer
           // Special case of a `cudaq::state*` which must be in the same address
           // space. This references a container to a set of simulation
           // amplitudes.
-          synthesizeRuntimeArgument<cudaq::state *>(
-              builder, argument, args, offset, sizeof(void *),
-              [=](OpBuilder &builder, cudaq::state **concrete) {
-                Value rawPtr = builder.create<arith::ConstantIntOp>(
-                    loc, reinterpret_cast<std::intptr_t>(*concrete),
-                    sizeof(void *) * 8);
-                auto stateTy = cudaq::cc::StateType::get(builder.getContext());
-                return builder.create<cudaq::cc::CastOp>(
-                    loc, cudaq::cc::PointerType::get(stateTy), rawPtr);
-              });
-          continue;
+          if (sameAddressSpace) {
+            synthesizeRuntimeArgument<cudaq::state *>(
+                builder, argument, args, offset, sizeof(void *),
+                [=](OpBuilder &builder, cudaq::state **concrete) {
+                  Value rawPtr = builder.create<arith::ConstantIntOp>(
+                      loc, reinterpret_cast<std::intptr_t>(*concrete),
+                      sizeof(void *) * 8);
+                  auto stateTy =
+                      cudaq::cc::StateType::get(builder.getContext());
+                  return builder.create<cudaq::cc::CastOp>(
+                      loc, cudaq::cc::PointerType::get(stateTy), rawPtr);
+                });
+            continue;
+          } else {
+            funcOp.emitOpError("synthesis: unsupported argument type for "
+                               "remote devices and simulators: state*");
```

Comment:
It might be nice to have a test to demonstrate/exercise this error message, perhaps like `targettests/execution/qir_range_err_runtime.cpp`, but updated for `remote-mqpu` or `nvqc`.

Reply:
will do, thanks!

Reply:
done!